domenetjenester/inngangsvilkar/src/main/java/no/nav/foreldrepenger/inngangsvilkaar/medlemskap/v2/MedlemskapsvilkårRegelGrunnlagBygger.java

```diff
@@ -0,0 +1,121 @@
+package no.nav.foreldrepenger.inngangsvilkaar.medlemskap.v2;
+
+import static no.nav.foreldrepenger.inngangsvilkaar.medlemskap.v2.MedlemskapsvilkårRegelGrunnlag.Personopplysninger.PersonstatusPeriode;
+import static no.nav.foreldrepenger.inngangsvilkaar.medlemskap.v2.MedlemskapsvilkårRegelGrunnlag.Personopplysninger.PersonstatusPeriode.Type;
+import static no.nav.foreldrepenger.inngangsvilkaar.medlemskap.v2.MedlemskapsvilkårRegelGrunnlag.Personopplysninger.Region;
+import static no.nav.foreldrepenger.inngangsvilkaar.medlemskap.v2.MedlemskapsvilkårRegelGrunnlag.Personopplysninger.RegionPeriode;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+import no.nav.foreldrepenger.behandling.BehandlingReferanse;
+import no.nav.foreldrepenger.behandlingslager.behandling.medlemskap.MedlemskapOppgittTilknytningEntitet;
+import no.nav.foreldrepenger.behandlingslager.behandling.personopplysning.PersonstatusEntitet;
+import no.nav.foreldrepenger.behandlingslager.geografisk.Landkoder;
+import no.nav.foreldrepenger.behandlingslager.geografisk.MapRegionLandkoder;
+import no.nav.foreldrepenger.domene.medlem.MedlemTjeneste;
+import no.nav.foreldrepenger.domene.personopplysning.PersonopplysningTjeneste;
+import no.nav.foreldrepenger.domene.tid.DatoIntervallEntitet;
+import no.nav.fpsak.tidsserie.LocalDateInterval;
+
+@ApplicationScoped
+class MedlemskapsvilkårRegelGrunnlagBygger {
+
+    private MedlemTjeneste medlemTjeneste;
+    private PersonopplysningTjeneste personopplysningTjeneste;
+
+    @Inject
+    MedlemskapsvilkårRegelGrunnlagBygger(MedlemTjeneste medlemTjeneste, PersonopplysningTjeneste personopplysningTjeneste) {
+        this.medlemTjeneste = medlemTjeneste;
+        this.personopplysningTjeneste = personopplysningTjeneste;
+    }
+
+    MedlemskapsvilkårRegelGrunnlagBygger() {
+        //CDI
+    }
+
+    MedlemskapsvilkårRegelGrunnlag lagRegelGrunnlag(BehandlingReferanse behandlingRef) {
+        var vurderingsperiode = utledVurderingsperiode(behandlingRef);
+        var registrertMedlemskapPerioder = hentMedlemskapPerioder(behandlingRef);
+        var personopplysningGrunnlag = hentPersonopplysninger(behandlingRef, vurderingsperiode);
+        var søknad = hentSøknad(behandlingRef);
+
+        return new MedlemskapsvilkårRegelGrunnlag(vurderingsperiode, registrertMedlemskapPerioder, personopplysningGrunnlag, søknad);
+    }
+
+    private LocalDateInterval utledVurderingsperiode(BehandlingReferanse behandlingRef) {
+        var fom = behandlingRef.skjæringstidspunkt().getUtledetSkjæringstidspunkt();
```

Comment:
1 dag? Bosatt vurderes innenfor 12m i Inngang. Vi skal ta en runder på lovligOpphold om den er 12m eller 1dag

Reply:
Lager to utledere utledVurderingsperiodeLovligOpphold og utledVurderingsperiodeBosatt og venter på integrasjon her.